app/controllers/application_controller.rb

```diff
@@ -94,6 +95,18 @@ def render_404
 
   private
 
+  def expire_oauth_session
+    if (session[:oauth_uid].present? || session[:oauth_provider].present?) && session[:oauth_expires_at].blank?
```

Comment:
Are these only set for the archive oauth?

Reply:
Currently yes, but just in case we need it later, I decided make it extensible. This particular method lives here because it was the most efficient way I could think of to have any request clear the state if it was made after expiration.